src/database/sql_database.cc

```diff
@@ -364,9 +364,14 @@ std::vector<std::shared_ptr<SQLDatabase::AddUpdateTable>> SQLDatabase::_addUpdat
 void SQLDatabase::addObject(std::shared_ptr<CdsObject> obj, int* changedContainer)
 {
     if (obj->getID() != INVALID_OBJECT_ID)
-        throw_std_runtime_error("tried to add an object with an object ID set");
+        throw_std_runtime_error("Tried to add an object with an object ID set");
 
-    std::vector<std::shared_ptr<SQLDatabase::AddUpdateTable>> tables = _addUpdateObject(obj, false, changedContainer);
+    if (config->getOption(CFG_SERVER_STORAGE_DRIVER) == "sqlite3") {
+        exec("BEGIN TRANSACTION");
+    } else {
+        exec("START TRANSACTION");
+    }
```

Comment:
You'd better add a new property beginTransaction and set it in constructor call - a virtual method seems a bit overdone.